config/config_test.go

```diff
@@ -70,6 +70,15 @@ func TestBaseConfigValidateBasic(t *testing.T) {
 	require.Error(t, cfg.ValidateBasic())
 }
 
+func TestBaseConfigValidateBasic2(t *testing.T) {
```

Comment:
Could we also test more examples here ? Like the `default` use case where we assume it's `tcp` ?